src/app/cli/src/init/graphql_client.ml

```diff
@@ -2,53 +2,99 @@ open Core
 open Async
 open Signature_lib
 
+let graphql_error_to_string e =
+  let error_obj_to_string obj =
+    let open Yojson.Basic in
+    let obj_message =
+      let open Option.Let_syntax in
+      let%bind message = Util.to_option (Util.member "message") obj in
+      let%map path = Util.to_option (Util.member "path") obj in
+      let message =
+        Util.to_string_option message
+        |> Option.value ~default:(Yojson.Basic.to_string message)
+      in
+      Printf.sprintf "%s (in %s)" message (Yojson.Basic.to_string path)
+    in
+    match obj_message with Some m -> m | None -> to_string obj
+  in
+  match e with
+  | `List l ->
+      List.map ~f:error_obj_to_string l |> String.concat ~sep:"\n"
+  | e ->
+      error_obj_to_string e
+
 let query_or_error
     (query_obj :
       < parse: Yojson.Basic.json -> 'response
       ; query: string
       ; variables: Yojson.Basic.json
-      ; .. >) port : 'response Deferred.Or_error.t =
+      ; .. >) port :
+    ( 'response
+    , [`Failed_request of string | `Graphql_error of string] )
+    Deferred.Result.t =
   let uri_string = "http://localhost:" ^ string_of_int port ^ "/graphql" in
   let variables_string = Yojson.Basic.to_string query_obj#variables in
   let body_string =
     Printf.sprintf {|{"query": "%s", "variables": %s}|} query_obj#query
       variables_string
   in
   let query_uri = Uri.of_string uri_string in
-  let open Deferred.Let_syntax in
-  let get_result () =
-    let%bind _, body =
-      Cohttp_async.Client.post
-        ~headers:
-          (Cohttp.Header.add (Cohttp.Header.init ()) "Accept"
-             "application/json")
-        ~body:(Cohttp_async.Body.of_string body_string)
-        query_uri
-    in
-    let%map body = Cohttp_async.Body.to_string body in
-    Yojson.Basic.from_string body
-    |> Yojson.Basic.Util.member "data"
-    |> query_obj#parse
+  let open Deferred.Result.Let_syntax in
+  let%bind _, body =
+    Deferred.Or_error.try_with ~extract_exn:true (fun () ->
+        Cohttp_async.Client.post
+          ~headers:
+            (Cohttp.Header.add (Cohttp.Header.init ()) "Accept"
+               "application/json")
+          ~body:(Cohttp_async.Body.of_string body_string)
+          query_uri )
+    |> Deferred.Result.map_error ~f:(fun e ->
+           `Failed_request (Error.to_string_hum e) )
+  in
+  let%bind body_str =
+    Cohttp_async.Body.to_string body |> Deferred.map ~f:Result.return
   in
-  Deferred.Or_error.try_with ~extract_exn:true get_result
+  let body_json = Yojson.Basic.from_string body_str in
+  let open Yojson.Basic.Util in
+  ( match (member "errors" body_json, member "data" body_json) with
+  | `Null, `Null ->
+      Error (`Graphql_error "Empty response from graphql query")
+  | error, `Null ->
+      Error (`Graphql_error (graphql_error_to_string error))
+  | _, data ->
+      Result.try_with (fun () -> query_obj#parse data)
+      |> Result.map_error ~f:(fun _ ->
```

Comment:
I think Graphql_error should should the error that we receive if we encounter an exception.

Reply:
Usually this exception is not very descriptive, but I added it for now:
```
❌ Error: Problem parsing graphql response
Error message: (Failure "graphql_ppx: Object is not a value")
```